astropy/nddata/utils.py

```diff
@@ -456,3 +469,322 @@ def block_replicate(data, block_size, conserve_sum=True):
+
+        # extract_array and overlap_slices use (y, x) positions
+        pos = position[::-1]
+
+        data = np.asanyarray(data)
+        cutout_data, input_position_cutout  = extract_array(
+            data, shape, pos, mode=mode, fill_value=fill_value,
+            return_position=True)
+        if copy:
+            cutout_data = np.copy(cutout_data)
+        self.data = cutout_data
+
+        self.input_position_cutout = input_position_cutout[::-1]    # (x, y)
+        slices_original, slices_cutout = overlap_slices(data.shape, shape,
+                                                    pos, mode=mode)
+        self.slices_original = slices_original
+        self.slices_cutout = slices_cutout
+
+        self.shape = self.data.shape
+        self.input_position_original = position
+        self.shape_input = shape
+
+        (self.ymin_original, self.xmin_original,
+         self.ymax_original, self.xmax_original) = self.bbox_original
+
+        (self.ymin_cutout, self.xmin_cutout,
+         self.ymax_cutout, self.xmax_cutout) = self.bbox_cutout
+
+        # the true origin pixel of the cutout array, including any
+        # filled cutout values
+        self._origin_original_true = (self.origin_original[0] -
+                                   self.slices_cutout[1].start,
+                                   self.origin_original[1] -
+                                   self.slices_cutout[0].start)
+
+        if wcs is not None:
+            self.wcs = deepcopy(wcs)
+            self.wcs.wcs.crpix -= self._origin_original_true
+        else:
+            self.wcs = None
+
+    def to_original_position(self, cutout_position):
+        """
+        Convert an ``(x, y)`` position in the cutout array to the original
+        ``(x, y)`` position in the original large array.
+
+        Parameters
+        ----------
+        cutout_position : tuple
+            The ``(x, y)`` pixel position in the cutout array.
+
+        Returns
+        -------
+        original_position : tuple
+            The corresponding ``(x, y)`` pixel position in the original
+            large array.
+        """
+        return tuple(cutout_position[i] + self.origin_original[i]
+                     for i in [0, 1])
+
+    def to_cutout_position(self, original_position):
+        """
+        Convert an ``(x, y)`` position in the original large array to
+        the ``(x, y)`` position in the cutout array.
+
+        Parameters
+        ----------
+        original_position : tuple
+            The ``(x, y)`` pixel position in the original large array.
+
+        Returns
+        -------
+        cutout_position : tuple
+            The corresponding ``(x, y)`` pixel position in the cutout
+            array.
+        """
+        return tuple(original_position[i] - self.origin_original[i]
+                     for i in [0, 1])
+
+    def plot_on_original(self, ax=None, fill=False, **kwargs):
```

Comment:
Maybe return `ax` to make it simpler for the user to continue drawing on that `ax` in case it was constructed in the method?